### PR TITLE
filter Announcements by course

### PR DESCRIPTION
**Pull request by AhmedElemary57**
Adding the filter to the announcements page completed back and front. 
UI is not that good but it works for now

### CODE REVIEW

File: Student-Unity-frontend/src/app/components/announcement/announcement.component.ts [hunk 1]
```diff
@@ -34,4 +34,8 @@ export class AnnouncementComponent implements OnInit {
         let uniqueId = randomLetter + Date.now();
         return uniqueId;
     }
+
+  removeTask() {
```
Comment (Ahmedelsa3eed) [inline Student-Unity-frontend/src/app/components/announcement/announcement.component.ts:38]: what's removeTask do here in announcement page

File: Student-Unity/src/main/java/csed/swe/studentunity/dao/AnnouncementRepo.java [hunk 2]
```diff
@@ -15,4 +17,15 @@ public interface AnnouncementRepo extends JpaRepository<Announcement, UUID> {
...
+    @Query(value =
+            "SELECT course_name, body, posted_date " +
+            "FROM announcement AS a JOIN registered_course AS rc ON a.course_id = rc.course_id " +
+                    "JOIN course AS c ON rc.course_id = c.course_id " +
+            "WHERE rc.id = ?1 AND a.course_id = ?2", nativeQuery = true)
+    Optional<List<Object>> filterAnnouncements(Long userId, Long courseId);
+
+    @Modifying
+    @Query( value = "DELETE FROM announcement WHERE id = ?1", nativeQuery = true)
+    void deleteById(Long announcementId);
```
Comment (Ahmedelsa3eed) [inline Student-Unity/src/main/java/csed/swe/studentunity/dao/AnnouncementRepo.java:30]: i think if you typed ``deleteAnnouncementById(Long announcementId);``, it should also work.

**Review by Ahmedelsa3eed — APPROVED** [review]

File: Student-Unity-frontend/src/app/components/announcement/announcement.component.ts (hunk 1 above)
Comment (AhmedElemary57) [reply Student-Unity-frontend/src/app/components/announcement/announcement.component.ts:38]: no thing I was planning to do remove announcement like remove  task  but I changed My idea.
I did it in the back only

File: Student-Unity/src/main/java/csed/swe/studentunity/dao/AnnouncementRepo.java (hunk 2 above)
Comment (AhmedElemary57) [reply Student-Unity/src/main/java/csed/swe/studentunity/dao/AnnouncementRepo.java:30]: I like to make the Queries natively And not rely on JPA auto-generated queries

File: Student-Unity-frontend/src/app/components/announcement/announcement.component.ts (hunk 1 above)
Comment (Ahmedelsa3eed) [reply Student-Unity-frontend/src/app/components/announcement/announcement.component.ts]: we may do it later

**Pull request merged**